ipatests/test_integration/test_membermanager.py

```diff
@@ -212,3 +212,51 @@ def test_hostgroup_member_manager_nopermission(self):
             "'write' privilege to the 'memberManager' attribute of entry"
         )
         assert expected in result.stdout_text
+
+    def test_member_manager_upgrade_scenario(self):
+        user = "idmuser"
+        password = "Secret123"
+        group1 = "role-groupmanager"
+        group2 = "role-usergroup-A"
+
+        master = self.master
+
+        tasks.group_add(master, group1)
+        tasks.group_add(master, group2)
+        tasks.create_active_user(master, user, password)
+
+        tasks.kinit_admin(master)
+        master.run_command(["ipa", "group-add-member", group1, "--user", user])
+        master.run_command(["ipa", "group-add-member-manager", "--groups",
+                            group1, group2])
+
+        ldif_text = """dn: cn=hostgroups,cn=accounts,dc=ipa,dc=test
+        changetype: modify
+        delete: aci
+        aci: (targetattr = "member")(targetfilter =
+        "(objectclass=ipaHostGroup)")(version 3.0; acl
+        "Allow member managers to modify members of host groups"; allow (write)
+        userattr = "memberManager#USERDN" or userattr =
+        "memberManager#GROUPDN";)
+
+        dn: cn=groups,cn=accounts,dc=ipa,dc=test
+        changetype: modify
+        delete: aci
+        aci: (targetattr = "member")(targetfilter =
+        "(objectclass=ipaUserGroup)")(version 3.0; acl
+        "Allow member managers to modify members of user groups"; allow (write)
+        userattr = "memberManager#USERDN" or userattr =
+        "memberManager#GROUPDN";)"""
+        tasks.ldapmodify_dm(master, ldif_text)
+
+        # should fail
+        # tasks.kinit_as_user(master, user, password)
+        # master.run_command(["ipa", "group-add-member" "--users",
+        # "admin", group2])
+
+        master.run_command(['ipa-server-upgrade'], raiseonerr=True)
+        try:
+            master.run_command(["ipa", "group-add-member", "--users",
+                                "admin", group2])
+        except Exception:
+            raise Exception("Exeption error: MemberManager with groups fails")
```

Comment:
you can use following instead of try except
```
result = master.run_command(
    ["ipa", "group-add-member", "--users", "admin", group2],
    raiseonerr=False
)
assert 'assert your error message here' in result.stdout_text
```

Reply:
Thanks for this, this was only WIP, it's reimplemented. Take a look, please.